code/lib/cli-storybook/src/automigrate/fixes/consolidated-imports.ts

```diff
@@ -0,0 +1,215 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+import { type NodePath, parser, recast, types as t, traverse } from 'storybook/internal/babel';
+import { commonGlobOptions, getProjectRoot } from 'storybook/internal/common';
+
+// eslint-disable-next-line depend/ban-dependencies
+import { globby } from 'globby';
+import pLimit from 'p-limit';
+import { dedent } from 'ts-dedent';
+
+import { type ConsolidatedPackage, consolidatedPackages } from '../helpers/consolidated-packages';
+import type { Fix, RunOptions } from '../types';
+
+export interface ConsolidatedOptions {
+  packageJsonFiles: string[];
+}
+
+function transformPackageJson(content: string): string | null {
+  const packageJson = JSON.parse(content);
+  let hasChanges = false;
+
+  // Check dependencies
+  if (packageJson.dependencies) {
+    for (const [dep, version] of Object.entries(packageJson.dependencies)) {
+      if (dep in consolidatedPackages) {
+        delete packageJson.dependencies[dep];
+        hasChanges = true;
+      }
+    }
+  }
+
+  // Check devDependencies
+  if (packageJson.devDependencies) {
+    for (const [dep, version] of Object.entries(packageJson.devDependencies)) {
+      if (dep in consolidatedPackages) {
+        delete packageJson.devDependencies[dep];
+        hasChanges = true;
+      }
+    }
+  }
+
+  // Ensure storybook is in devDependencies
+  if (!packageJson.devDependencies) {
+    packageJson.devDependencies = {};
+  }
+  if (!packageJson.devDependencies.storybook) {
+    packageJson.devDependencies.storybook = '^8.0.0';
+    hasChanges = true;
+  }
```

Comment:
In Storybook 8, `storybook` was already required to be present in the user's package.json. It's not necessary to check for it. Additionally, it can also add `storybook` unnecessarily in nested package.jsons, especially in mono repos where NX's single version policy is applied, only the root package.json contains dependencies, whereas leaf package.jsons only defines scripts.